### DISCUSSION TITLE
Plotting stacked spectra with different grids

### DISCUSSION THREAD

**Discussion by sjgunderson** [sherpa/sherpa#1951] [Ideas]
Would it be possible in Sherpa to match the grids of different spectra on the same object, say Chandra's HEG and MEG, and plot the sum instead of an individual plot for each one? This can be done in ISIS and would be a great addition for Sherpa.

**Comment by hamogu** [comment]
@sjgunderson : This would fit naturally into the work that I plan on doing around the current datastack module soon. I know @DougBurke also has prototype of something at least for plotting, so I hope we can get something into Sherpa itself in the next release.

**Comment by DougBurke** [comment]
I am a little unclear on what people say when they talk about this sort of thing (so, it's not this particular question). Is it being able to some up the `plot_source` plots or `plot_model` / `plot_fit` plots or something else?

**Comment by hamogu** [comment]
In grating data in particular the signal is often very low, but there are specific sharp features at a known position. So, you might have 2 ct in HEG+1, 1 ct in HEG-1, 2 cts in MEG+1 etc  in the line with 0-1 cts (usually 0) / bin in the surrounding bins from background and source continuum.
If you just plot all 4 spectra and their models on top if each other, you get four lines jumping between 0 and 2 and the emission line won't stand out and the plot will generally be visually terrible. However, if you coadd all the data (for plotting, not for fitting because of different RMFs etc) you will clearly see the line with (in my example above) 6 cts/ bin where the continuum/background around it will be 0-1 cts/bin.

My example is for really faint spectra, but the same holds true for brighter lines, where co-adding all HEG and MEG orders gives you about 4 time the signal than an individual spectrum does and plotting 4 spectra on top of each other never looks good.

I'm mentioning grating spectra in particular, because they have many sharp features, so the lines in the plotted spectrum jump up and down a lot unlike ACIS or XMM/EPIC where you typically have several bins per resolution element and the plots look visually "smoother" (not a good thing, because it mean lower resolution!) and thus are easier to follow with your eyes.

Next time I plot a grating spectrum, I'll fake up an example and add it to this thread, but I don't have one handy right now and don't want to get into doign that right now because I have a new (non Sherpa) things to finish with deadlines.

In order to co-add, the bins need to be on the same wavelength grid.

**Comment by hamogu** [comment]
So, this would be for `plot_fit` or `plot_data`.

**Comment by DougBurke** [comment]
So, the models are in channels (for the X axis) which makes it easy to add up, but you (maybe) need to worry about different grouping choices for the different datasets (i.e. what grouping should the sum be). Now, we have the odd behaviour of `plot_fit` vs `plot_model` [*], which is something I'd like to address,but that's separate from this.

[*] that is, `plot_model` does not group the model but `plot_fit` does - we should be able to decide for each plot type whether to apply the grouping or not for PHA data.

**Comment by DougBurke** [comment]
See some discussion in #2425
